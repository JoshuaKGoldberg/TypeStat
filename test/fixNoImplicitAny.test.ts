import path from "node:path";
import { describe, expect, it } from "vitest";

import { runMutationTest } from "../src/tests/testSetup.js";

describe("noImplicitAny", () => {
	it("parameters", async () => {
		const caseDir = path.join(
			import.meta.dirname,
			"cases/fixes/noImplicitAny/parameters",
		);
		const { actualContent, expectedFilePath } = await runMutationTest(caseDir);
		await expect(actualContent).toMatchFileSnapshot(expectedFilePath);
	}, 6000);

	it("property declarations", async () => {
		const caseDir = path.join(
			import.meta.dirname,
			"cases/fixes/noImplicitAny/propertyDeclarations",
		);
		const { actualContent, expectedFilePath } = await runMutationTest(caseDir);
		await expect(actualContent).toMatchFileSnapshot(expectedFilePath);
	});

	it("variable declarations", async () => {
		const caseDir = path.join(
			import.meta.dirname,
			"cases/fixes/noImplicitAny/variableDeclarations",
		);
		const { actualContent, expectedFilePath } = await runMutationTest(caseDir);
		await expect(actualContent).toMatchFileSnapshot(expectedFilePath);
	});
});
