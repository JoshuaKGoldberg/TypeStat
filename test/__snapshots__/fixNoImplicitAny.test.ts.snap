// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`noImplicitAny > parameters > options 1`] = `
"[
  {
    "cleanups": {
      "suppressTypeErrors": true
    },
    "compilerOptions": {
      "noImplicitAny": true,
      "noImplicitThis": false,
      "strictNullChecks": true
    },
    "files": {
      "above": "",
      "below": "",
      "renameExtensions": false
    },
    "filters": [],
    "fixes": {
      "importExtensions": false,
      "incompleteTypes": false,
      "missingProperties": false,
      "noImplicitAny": true,
      "noImplicitThis": false,
      "noInferableTypes": false,
      "strictNonNullAssertions": false
    },
    "hints": {
      "react": {
        "propTypes": "whenRequired",
        "propTypesOptionality": "asWritten"
      }
    },
    "mutators": [
      [
        "fixImportExtensions",
        null
      ],
      [
        "fixIncompleteTypes",
        null
      ],
      [
        "fixMissingProperties",
        null
      ],
      [
        "fixNoImplicitAny",
        null
      ],
      [
        "fixNoImplicitThis",
        null
      ],
      [
        "fixNoInferableTypes",
        null
      ],
      [
        "fixStrictNonNullAssertions",
        null
      ]
    ],
    "output": {},
    "package": {
      "directory": "<rootDir>",
      "file": "<rootDir>/package.json"
    },
    "postProcess": {
      "shell": []
    },
    "projectPath": "<rootDir>/tsconfig.json",
    "types": {
      "strictNullChecks": true
    }
  }
]"
`;

exports[`noImplicitAny > property declarations > options 1`] = `
"[
  {
    "cleanups": {
      "suppressTypeErrors": true
    },
    "compilerOptions": {
      "noImplicitAny": false,
      "noImplicitThis": false,
      "strictNullChecks": true
    },
    "files": {
      "above": "",
      "below": "",
      "renameExtensions": false
    },
    "filters": [],
    "fixes": {
      "importExtensions": false,
      "incompleteTypes": true,
      "missingProperties": false,
      "noImplicitAny": false,
      "noImplicitThis": false,
      "noInferableTypes": false,
      "strictNonNullAssertions": false
    },
    "hints": {
      "react": {
        "propTypes": "whenRequired",
        "propTypesOptionality": "asWritten"
      }
    },
    "mutators": [
      [
        "fixImportExtensions",
        null
      ],
      [
        "fixIncompleteTypes",
        null
      ],
      [
        "fixMissingProperties",
        null
      ],
      [
        "fixNoImplicitAny",
        null
      ],
      [
        "fixNoImplicitThis",
        null
      ],
      [
        "fixNoInferableTypes",
        null
      ],
      [
        "fixStrictNonNullAssertions",
        null
      ]
    ],
    "output": {},
    "package": {
      "directory": "<rootDir>",
      "file": "<rootDir>/package.json"
    },
    "postProcess": {
      "shell": []
    },
    "projectPath": "<rootDir>/tsconfig.json",
    "types": {
      "strictNullChecks": true
    }
  }
]"
`;

exports[`noImplicitAny > variable declarations > options 1`] = `
"[
  {
    "cleanups": {
      "suppressTypeErrors": true
    },
    "compilerOptions": {
      "noImplicitAny": false,
      "noImplicitThis": false,
      "strictNullChecks": true
    },
    "files": {
      "above": "",
      "below": "",
      "renameExtensions": false
    },
    "filters": [],
    "fixes": {
      "importExtensions": false,
      "incompleteTypes": true,
      "missingProperties": false,
      "noImplicitAny": false,
      "noImplicitThis": false,
      "noInferableTypes": false,
      "strictNonNullAssertions": false
    },
    "hints": {
      "react": {
        "propTypes": "whenRequired",
        "propTypesOptionality": "asWritten"
      }
    },
    "mutators": [
      [
        "fixImportExtensions",
        null
      ],
      [
        "fixIncompleteTypes",
        null
      ],
      [
        "fixMissingProperties",
        null
      ],
      [
        "fixNoImplicitAny",
        null
      ],
      [
        "fixNoImplicitThis",
        null
      ],
      [
        "fixNoInferableTypes",
        null
      ],
      [
        "fixStrictNonNullAssertions",
        null
      ]
    ],
    "output": {},
    "package": {
      "directory": "<rootDir>",
      "file": "<rootDir>/package.json"
    },
    "postProcess": {
      "shell": []
    },
    "projectPath": "<rootDir>/tsconfig.json",
    "types": {
      "strictNullChecks": true
    }
  }
]"
`;
